sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -712,6 +713,7 @@ def __init__(
     self.bq_io_metadata = None  # Populate in setup, as it may make an RPC
     self.bigquery_job_labels = bigquery_job_labels or {}
     self.use_json_exports = use_json_exports
+    self.temp_dataset = temp_dataset
```

Comment:
how about we add a `get_temporary_dataset` function to `BigqueryWrapper`, and we can define a `self.temp_dataset = self.temp_dataset or bq.get_temporary_dataset()` (this logic would need to run in the `split`, and `- and once we've done that, we can just pass the dataset name around to every call? That way we will treat the user-defined dataset and the automatic dataset the same way?

Reply:
let me know what you think about this implementation @frankzhao

I understand changing it may be troublesome, but doing this would simplify the logic quite a bit

Reply:
I agree, let me take look at it